File: app/src/main/java/com/example/servicebookingandroid/Request/Comment/CommentAdapter.java
package com.example.servicebookingandroid.Request.Comment;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.example.servicebookingandroid.Model.CommentDto;
import com.example.servicebookingandroid.Model.UserDto;
import com.example.servicebookingandroid.R;

import java.util.List;

public class CommentAdapter extends ArrayAdapter<CommentDto> {
    private List<CommentDto> commentDtos;
    private Context context;

    public CommentAdapter(@NonNull Context context, int resource, @NonNull List<CommentDto> objects) {
        super(context, resource, objects);
        this.commentDtos = objects;
        this.context = context;
    }

    @NonNull
    @Override
    public View getView(final int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        LayoutInflater layoutInflater =(LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View view=layoutInflater.inflate(R.layout.list_view_comment_item, parent,false);

        TextView tv_name = view.findViewById(R.id.tv_name);
        TextView tv_detail = view.findViewById(R.id.tv_detail);
        TextView tv_location = view.findViewById(R.id.tv_location);
        TextView tv_phone = view.findViewById(R.id.tv_phone);
        TextView tv_language = view.findViewById(R.id.tv_language);

        CommentDto commentDto = commentDtos.get(position);
        UserDto userDto = commentDto.getUserdto();

        String name = userDto.getFirstname() + " " + userDto.getLastname();
        String detail = commentDto.getCommentDetail();
        String location = userDto.getStreetname() + ", " + userDto.getCity() + ", " + userDto.getState() + ", " + userDto.getZipcode();
        String phone = userDto.getPhone();
        String language = userDto.getLanguage();

        tv_name.setText("@" + name);
        tv_detail.setText(detail);
        tv_location.setText("Location: " + location);
        tv_phone.setText("Phone: " + phone);
        tv_language.setText("Language: " + language);

        return view;
    }
}

File: app/src/main/java/com/example/servicebookingandroid/Request/Comment/CommentAdapter.kt
package com.example.servicebookingandroid.Request.Comment

import android.content.Context
import android.text.Layout
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.example.servicebookingandroid.Model.CommentDto
import com.example.servicebookingandroid.R

public class CommentAdapter: ArrayAdapter<CommentDto> {
    lateinit var commentDtos: List<CommentDto>
    constructor(context: Context, resource: Int, objects: List<CommentDto>):super(context, resource, objects){
        commentDtos = objects
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var layoutInflater: LayoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        var view: View = layoutInflater.inflate(R.layout.list_view_comment_item, parent,false)

        val tv_name = view.findViewById<TextView>(R.id.tv_name)
        val tv_detail = view.findViewById<TextView>(R.id.tv_detail)
        val tv_location = view.findViewById<TextView>(R.id.tv_location)
        val tv_phone = view.findViewById<TextView>(R.id.tv_phone)
        val tv_language = view.findViewById<TextView>(R.id.tv_language)

        val commentDto = commentDtos[position]
        var userDto = commentDto.userdto

        val name = userDto.firstname + " " + userDto.lastname
        val detail = commentDto.commentDetail
        val location = userDto.streetname + ", " + userDto.city + ", " + userDto.state + ", " + userDto.zipcode
        val phone = userDto.phone
        val language = userDto.language

        tv_name.text = "@$name"
        tv_detail.text = detail
        tv_location.text = "Location: $location"
        tv_phone.text = "Phone: $phone"
        tv_language.text = "Language: $language"

        return view
    }
}